pkg/cmd/cli/cmd/project.go

```diff
@@ -199,13 +212,14 @@ func (o ProjectOptions) RunProject() error {
 
 		namespaceInUse = project.Name
 		contextInUse = merged.CurrentContext
+		isDefaultContext = true
 	}
 
 	if err := kubecmdconfig.ModifyConfig(o.PathOptions, config); err != nil {
 		return err
 	}
 
-	if contextInUse != namespaceInUse {
+	if contextInUse != namespaceInUse && !isDefaultContext {
```

Comment:
Users already know who they are (they logged in as themselves) so the information in the context isn't relevant.  For normal happy path they don't care about that, so including it confuses them (thinks they did something wrong).  Reward the user with simple terse statements when they do the right thing (keep their mental model), explain to them when they step out of bounds by being more verbose.  That means they recognize "tons of gorp" as "Hrm, I did something wrong so now I need to read what they sent".

----- Original Message -----
> >  	}
> >  
> >  	if err := kubecmdconfig.ModifyConfig(o.PathOptions, config); err != nil {
> >  		return err
> >  	}
> >  
> > -	if contextInUse != namespaceInUse {
> > +	if contextInUse != namespaceInUse && !isDefaultContext {
> 
> Why not unconditionally output, "Now using project %s on server %s as user
> %s"?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2405/files#r30794302